<!-- Launch file for starting the openai panda simulator -->
<launch>

    <!--## ARGUMENTS ##-->
    <!-- Simulation arguments -->
    <arg name="paused" default="false" />
    <arg name="moveit" default="true" />
    <arg name="world_name" default="assets/worlds/empty.world" />
    <arg name="rviz" default="false" />
    <arg name="moveit_rviz_file" default="$(find panda_training)/launch/panda.rviz" />
    <arg name="load_gripper" default="true" />
    <arg name="grip_site" default="true" />
    <arg name="grip_site_xyz" default="0.0 0.0 0.10" />
    <arg name="grip_site_rpy" default="0.0 0.0 0.0" />

    <!-- ## LOAD ROBOT DESCRIPTION ## -->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find franka_description)/robots/panda_arm_hand.urdf.xacro" />

    <!--## LOAD CONTROLLERS ##
        NOTE: Make sure the load_controllers argument is set to false if this block is uncommented -->
    <include file="$(find panda_training)/launch/controllers.launch">
        <arg name="load_gripper" value="$(arg load_gripper)" />
    </include>

    <!--## LOAD MOVEIT CONTROL SERVICES ##-->
    <node if="$(arg moveit)" pkg="panda_training" type="panda_moveit_server.py" name="panda_moveit_planner_server" />
    <!-- <node pkg="panda_training" type="panda_control_server.py" name="panda_control_server" /> -->

    <!--## SIMULATION ##-->
    <!-- Launch simulation -->
    <include file="$(find panda_simulation)/launch/simulation.launch">
        <arg name="paused" value="$(arg paused)" />
        <arg name="rviz" value="$(arg rviz)" />
        <arg name="moveit" value="$(arg moveit)" />
        <arg name="world_name" value="$(arg world_name)" />
        <arg name="moveit_rviz_file" value="$(arg moveit_rviz_file)" />
        <arg name="load_gripper" value="$(arg load_gripper)" />
        <arg name="load_robot_state_publisher" default="true" />
        <arg name="load_controllers" value="false" />
        <arg name="grip_site" value="$(arg grip_site)" />
        <arg name="grip_site_xyz" value="$(arg grip_site_xyz)" />
        <arg name="grip_site_rpy" value="$(arg grip_site_rpy)" />
    </include>

    <!-- ## LOAD RVIZ ##
        NOTE: Only moveit was not used -->
    <include if="$(eval not arg('moveit') and arg('rviz'))" file="$(find panda_training)/launch/panda_rviz.launch" />
</launch>